osf/external/spam/tasks.py

```diff
@@ -33,18 +35,42 @@ def reclassify_domain_references(notable_domain_id, current_note, previous_note)
 def check_resource_for_domains(guid, content):
     from osf.models import Guid, NotableDomain, DomainReference
     resource = Guid.load(guid).referent
-    domains = {match.group('domain') for match in re.finditer(DOMAIN_REGEX, content)}
     spammy_domains = []
     referrer_content_type = ContentType.objects.get_for_model(resource)
-    for domain in domains:
-        domain, _ = NotableDomain.objects.get_or_create(domain=domain)
-        if domain.note == NotableDomain.Note.EXCLUDE_FROM_ACCOUNT_CREATION_AND_CONTENT:
-            spammy_domains.append(domain.domain)
+    for domain in _extract_domains(content):
+        notable_domain, _ = NotableDomain.objects.get_or_create(domain=domain)
+        if notable_domain.note == NotableDomain.Note.EXCLUDE_FROM_ACCOUNT_CREATION_AND_CONTENT:
+            spammy_domains.append(notable_domain.domain)
         DomainReference.objects.get_or_create(
-            domain=domain,
+            domain=notable_domain,
             referrer_object_id=resource.id,
             referrer_content_type=referrer_content_type,
-            defaults={'is_triaged': domain.note != NotableDomain.Note.UNKNOWN}
+            defaults={'is_triaged': notable_domain.note != NotableDomain.Note.UNKNOWN}
         )
     if spammy_domains:
         resource.confirm_spam(save=True, domains=list(spammy_domains))
+
+def _extract_domains(content):
+    extracted_domains = set()
+    for match in DOMAIN_REGEX.finditer(content):
+        domain = match.group('domain')
+        if not domain or domain in extracted_domains:
+            continue
+
+        protocol = match.group('protocol') or 'https://'
+        www = match.group('www') or ''
+        constructed_url = f'{protocol}{www}{domain}'
+
+        try:
+            response = requests.head(constructed_url)
+        except requests.exceptions.ConnectionError:
```

Comment:
Is this the only possible exception when invoking `requests.head`? What if it timesout, is it also a `ConnectionError`?

Reply:
That'd work. Should we also add a `timeout` parameter when invoking `requests.head()` so that it doesn't wait longer than necessary.